lib/drives/models/drive.dart

```diff
@@ -136,6 +132,16 @@ class Drive extends Trip {
   Future<void> cancel() async {
     cancelled = true;
     await supabaseClient.from('drives').update({'cancelled': true}).eq('id', id);
-    await supabaseClient.from('rides').update({'status': RideStatus.cancelledByDriver.index}).eq('drive_id', id);
+    //cancel all pending and approved rides for this drive.
+    await supabaseClient
+        .from('rides')
+        .update({'status': RideStatus.cancelledByDriver.index})
+        .eq('drive_id', id)
+        .or('status.eq.${RideStatus.pending.index},status.eq.${RideStatus.approved.index}');
```

Comment:
why "or"? I don't get ths

Reply:
So that only rides get updated that either are approved or pending.

Reply:
Ahhhh - So the "or" is just applied between the two conditions inside, and it is connected to the other condition via "and"? That's confusing...